### DISCUSSION TITLE
How to change report default severity?

### DISCUSSION THREAD

**Discussion by guuilp** [detekt/detekt#3942] [Q&A] [answered]
We're adopting Detekt on our project, where we already have Danger. Danger has a Detekt plugin that consumes the `xml` report from Detekt.

We want to treat all default Detekt rules as `error`. Even though it breaks the build, the `severity` are shown as `warning` on the report . Here's an example:

```
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="4.3">
  <file name="/Users/guilherme.pereira/Documents/Projects/consumer/feature/internal-settings/impl/src/main/java/InternalSettingsFragment.kt">
   <error line="51" column="38" severity="warning" message="Trailing space(s)" source="detekt.NoTrailingSpaces" />
  </file>
</checkstyle>
``` 

By reading the docs, it seems like this is the expected behavior: 

> The severity will be computed in the priority order:
> - Severity of the rule if exists
> - Severity of the parent ruleset if exists
> - Default severity: warning

https://detekt.github.io/detekt/reporting.html#severity

We've also seen [this great PR](https://github.com/detekt/detekt/pull/3310) that allows changing the severity per rule on the detekt config file. So the only option that we found is to manually set `severity: error` in all detekt rules on the config file, but this does not scale.

So the question is: Is there a way to change the report default severity (via config file, maybe)?

If the answer is no, we'd like to contribute if you think it makes sense.

**Comment by schalkms** [comment, marked as answer]
> So the only option that we found is to manually set severity: error in all detekt rules on the config file, but this does not scale.

It is enough if you set the severity once per rule set.
You don't need to set the severity for each and every rule.

**Reply by guuilp** [reply]
This solution works way better indeed!

But wouldn't it be cool to have control over the default severity too?

**Reply by picklebento** [reply]
Contributions are welcome for specifying the global default severity.

**Reply by schalkms** [reply]
> But wouldn't it be cool to have control over the default severity too?

What do you mean by that? You can actually change the severity in the config file. What do you find missing with the currently available feature set?

**Reply by guuilp** [reply]
Right now, Detekt has 10 rule sets. If I want to treat all rules with severity error, I need to change at least 10 lines.

What I wanted was to change just one line and affect all rules, so the default severity would be error and then I change some specific rules to warning.

**Reply by picklebento** [reply]
I would imagine the implementation would be under the config `output-reports:`

```
output-reports:
  active: true
  defaultSeverity: error
```

**Reply by guuilp** [reply]
Thanks for the suggestion!

I'll study detekt code and try to raise a PR.
